tests/e2e_tests/subcommands/weights/test_commit_weights.py

```diff
@@ -0,0 +1,179 @@
+        wait_for_finalization=True,
+        prompt=False,
+    )
+    assert result, "Failed to set weights rate limit"
+
+    # Configure the CLI arguments for the CommitWeightCommand
+    exec_command(
+        CommitWeightCommand,
+        [
+            "wt",
+            "commit",
+            "--no_prompt",
+            "--netuid",
+            "1",
+            "--uids",
+            str(uid),
+            "--weights",
+            str(weights),
+            "--subtensor.network",
+            "local",
+            "--subtensor.chain_endpoint",
+            "ws://localhost:9945",
+            "--wallet.path",
+            "/tmp/btcli-wallet",
+        ],
+    )
+
+    weight_commits = subtensor.query_module(
+        module="SubtensorModule",
+        name="WeightCommits",
+        params=[1, wallet.hotkey.ss58_address],
+    )
+
+    # Assert that the committed weights are set correctly
+    assert weight_commits.value is not None, "Weight commit not found in storage"
+    commit_hash, commit_block = weight_commits.value
+    assert commit_block > 0, f"Invalid block number: {commit_block}"
+
+    # Query the WeightCommitRevealInterval storage map
+    weight_commit_reveal_interval = subtensor.query_module(
+        module="SubtensorModule", name="WeightCommitRevealInterval", params=[1]
+    )
+    interval = weight_commit_reveal_interval.value
+    assert interval > 0, "Invalid WeightCommitRevealInterval"
+
+    # Wait until the reveal block range
+    current_block = subtensor.get_current_block()
+    reveal_block_start = (commit_block - (commit_block % interval)) + interval
+    while current_block < reveal_block_start:
+        time.sleep(1)  # Wait for 1 second before checking the block number again
+        current_block = subtensor.get_current_block()
+        if current_block % 10 == 0:
+            print(f"Current Block: {current_block}  Revealing at: {reveal_block_start}")
+
+    # Configure the CLI arguments for the RevealWeightCommand
+    exec_command(
+        RevealWeightCommand,
+        [
+            "wt",
+            "reveal",
+            "--no_prompt",
+            "--netuid",
+            "1",
+            "--uids",
+            str(uid),
+            "--weights",
+            str(weights),
+            "--subtensor.network",
+            "local",
+            "--subtensor.chain_endpoint",
+            "ws://localhost:9945",
+            "--wallet.path",
+            "/tmp/btcli-wallet",
+        ],
+    )
+
+    # Query the Weights storage map
+    revealed_weights = subtensor.query_module(
+        module="SubtensorModule", name="Weights", params=[1, uid]  # netuid and uid
+    )
+
+    # Assert that the revealed weights are set correctly
+    assert revealed_weights.value is not None, "Weight reveal not found in storage"
+
+    uid_list = list(map(int, re.split(r"[ ,]+", str(uid))))
+    uids = np.array(uid_list, dtype=np.int64)
+    weight_list = list(map(float, re.split(r"[ ,]+", str(weights))))
```

Comment:
As part of Roman's refactoring efforts, we've been trying to get away from `list(map(...`, and instead use comprehensions, so `[int(x) for x in re.split(r"[ ,]+", uid_str)]` for increased readability.

Reply:
+1